lxd/apparmor.go

```diff
@@ -384,7 +384,8 @@ profile "%s" flags=(attach_disconnected,mediate_deleted) {
 }
 
 func runApparmor(command string, c container) error {
-	if !aaAvailable {
+	state := c.StateObject()
```

Comment:
StateObject in containerLXC should probably be renamed to DaemonState() to more accurately represent what it is.

The comment under StateObject() which says "This function should go away" probably should also be removed at that point since you're adding new dependencies on it all over the place :)

Reply:
So for a bit of background, the StateObject() method has replaced a former Daemon() that used to return the full Daemon, see the relevant [diff](https://github.com/lxc/lxd/commit/2fe71e31137435e2c172781a89c0f780c7ffeb41#diff-35636ce597d0980ae9eefbf9c7d88b26L499). So from that point of view it's an improvement in that it reduces the coupling and dependency surface of the container interface. Ideally, I think we should eventually further reduce this coupling and drop StateObject() as well, but it's simply not possible at the moment because many higher-level APIs use container variables as "implicit handles" to database/OS state.

For these reasons I think the FIXME still makes sense, despite the fact that introducing new call sites invoking StateObject() does not help the eventual goal.

I renamed it DaemonObject() as you suggested and added some of the narrative above to the FIXME comment. See b3062c69557a3580936588d4fb8b46c49db227c4.